### PR TITLE
Parse fix

### PR DESCRIPTION
**Pull request by sykorjan**
Bug fixes related to issues https://issues.merck.com/browse/HLYD-265 and https://issues.merck.com/browse/HLYD-266.

### CODE REVIEW

File: tools/src/main/java/com/msd/gin/halyard/tools/HttpSparqlHandler.java [hunk 1]
```diff
@@ -187,9 +188,19 @@ private SparqlQuery retrieveQuery(HttpExchange exchange) throws IOException, Ill
         }
         if ("GET".equalsIgnoreCase(requestMethod)) {
             // Retrieve from the request URI parameter query and optional parameters defaultGraphs and namedGraphs
-            String requestQuery = exchange.getRequestURI().getQuery();
-            if (requestQuery != null) {
-                StringTokenizer stk = new StringTokenizer(requestQuery, AND_DELIMITER);
+            // Cannot apply directly exchange.getRequestURI().getQuery() since getQuery() method
+            // automatically decodes query (requestQuery must remain unencoded due to parsing by '&' delimiter)
+            String requestQueryRaw = exchange.getRequestURI().toString();
```
Comment (asotona) [inline tools/src/main/java/com/msd/gin/halyard/tools/HttpSparqlHandler.java]: wouldn't exchange.getRequestURI().getRawQuery() work here?

File: tools/src/main/java/com/msd/gin/halyard/tools/HttpSparqlHandler.java [hunk 2]
```diff
@@ -233,9 +244,19 @@ private SparqlQuery retrieveQuery(HttpExchange exchange) throws IOException, Ill
                     }
                 } else if (baseType.equals(UNENCODED_CONTENT)) {
                     // Retrieve from the request URI optional parameters defaultGraphs and namedGraphs
-                    String requestQuery = exchange.getRequestURI().getQuery();
-                    if (requestQuery != null) {
-                        StringTokenizer stk = new StringTokenizer(requestQuery, AND_DELIMITER);
+                    // Cannot apply directly exchange.getRequestURI().getQuery() since getQuery() method
+                    // automatically decodes query (requestQuery must remain unencoded due to parsing by '&' delimiter)
+                    String requestQueryRaw = exchange.getRequestURI().toString();
```
Comment (asotona) [inline tools/src/main/java/com/msd/gin/halyard/tools/HttpSparqlHandler.java]: and here .getRawQuery() as well ? :)

File: tools/src/main/java/com/msd/gin/halyard/tools/HttpSparqlHandler.java (hunk 1 above)
Comment (sykorjan) [reply tools/src/main/java/com/msd/gin/halyard/tools/HttpSparqlHandler.java]: It definitely would :)

File: tools/src/main/java/com/msd/gin/halyard/tools/HttpSparqlHandler.java (hunk 2 above)
Comment (sykorjan) [reply tools/src/main/java/com/msd/gin/halyard/tools/HttpSparqlHandler.java]: here as well :)

**Pull request merged**